script.js

```diff
@@ -1,4 +1,238 @@
+// -------------------------------------------------------------------
+
+//Dice Roll Function
+var myRandomValue = Math.random();
+var getRandomInteger = function (max) {
+  var randomDecimal = Math.random() * (max + 1);
+  var resultInteger = Math.floor(randomDecimal);
+  return resultInteger;
+};
+
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+//Display Running Score Function
+var getRunningScore = function () {
+  var runningScoreOutput = "Player 1 vs Player 2 Score History: ";
+  for (
+    var runningScoreCounter = 0;
+    runningScoreCounter < P2diceResultOrderedHistory.length;
+    runningScoreCounter += 1
+  ) {
+    runningScoreOutput =
+      runningScoreOutput +
+      "<br>" +
+      (runningScoreCounter + 1) +
+      ") " +
+      P1diceResultOrderedHistory[runningScoreCounter] +
+      " vs " +
+      P2diceResultOrderedHistory[runningScoreCounter];
+    +"<br>";
+  }
+  return runningScoreOutput;
+};
+
+// Display Leaderboard Function
+var getLeaderboard = function () {
+  var leaderboardOutput = "Leaderboard: <br>";
+  if (P1winCount > P2winCount) {
+    leaderboardOutput =
+      leaderboardOutput +
+      "P1 is in the lead with " +
+      P1winCount +
+      " win vs " +
+      P1loseCount +
+      " loss!<br> P2 is trailing behind with " +
+      P2winCount +
+      " win vs " +
+      P2loseCount +
+      " loss!<br>";
+  } else if (P2winCount > P1winCount) {
+    leaderboardOutput =
+      leaderboardOutput +
+      "P2 is in the lead with " +
+      P2winCount +
+      " win vs " +
+      P2loseCount +
+      " loss!<br> P1 is trailing behind with " +
+      P1winCount +
+      " win vs " +
+      P1loseCount +
+      " loss!<br>";
+  } else if (P1winCount == P2winCount) {
+    leaderboardOutput =
+      leaderboardOutput +
+      "Both players are currently tied at " +
+      P1winCount +
+      " win and " +
+      P1loseCount +
+      " loss each!";
+  }
+  return leaderboardOutput;
+};
+
+// -------------------------------------------------------------------
+// ----------------------- MAIN FUNCTION -----------------------------
+// -------------------------------------------------------------------
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (currentMode == "New Game") {
+    var P1dice1 = rollDice();
+    var P1dice2 = rollDice();
+    var myOutputValue =
+      "Welcome Player 1.<br> You rolled " +
+      P1dice1 +
+      " for Dice 1 and " +
+      P1dice2 +
+      " for Dice 2.<br> Choose the order of the dice by typing in which dice number (1 or 2) you want to go first.";
+    //Store dice results in global array
+    P1diceResult = [P1dice1, P1dice2];
+    console.log("P1 dice results");
+    console.log(P1diceResult);
+    //Change current mode
+    currentMode = "P1 Pick Order";
+  } else if (currentMode == "P1 Pick Order") {
```

Comment:
Good use of `Else if ` to 1. Give you control over the code flow, and 2. signal to the reader that the blocks of code are mutually exclusive